frame/vamm/src/lib.rs

```diff
@@ -1143,15 +1214,29 @@ pub mod pallet {
 				ensure!(!quote_twap.is_zero(), Error::<T>::NewTwapValueIsZero);
 			}
 
-			// 2) Vamm must be open.
+			// 2) Twaps must be reciprocal.
+			if let (Some(base_twap), Some(quote_twap)) = (base_twap, quote_twap) {
+				ensure!(
+					Self::twaps_are_reciprocal(base_twap, quote_twap)?,
+					Error::<T>::TwapsMustBeReciprocals
+				);
+			};
+
+			// 3) Vamm must be open.
 			ensure!(!Self::is_vamm_closed(vamm_state, now), Error::<T>::VammIsClosed);
 
-			// 3) Only update asset's twap if time has passed since last update.
+			// 4) Only update asset's twap if time has passed since last update
+			// and if the delay is greater than the required time between
+			// updates.
 			let now = Self::now(now);
-			ensure!(now > vamm_state.twap_timestamp, Error::<T>::AssetTwapTimestampIsMoreRecent);
+			ensure!(
+				now > vamm_state.twap_timestamp.try_add(&vamm_state.twap_period)?,
+				Error::<T>::AssetTwapTimestampIsMoreRecent
+			);
```

Comment:
Why would at least `twap_period` have to pass between updates? Wouldn't `weight_now` in line 1031 always be `twap_period` and `weight_last_twap` in line 1032 always be `1` if we force that?

For reference, this is how I use `twap_period` for the Oracle in the Clearing House
```
/// The reference time span used for weighting the EMA updates for the Oracle and Vamm TWAPs.
/// ```text
///                                                       twap_period
///                                        |---------------------------------------|
///                                         from_start          since_last
///                                        |-----------|---------------------------|
/// -------------------------------------------------------------------------------|
///                                        ^           ^                           ^
///                                      now -         |                          now
///                                   twap_period      |
///                                               last_twap_ts
/// ```
/// In the example above, the current price is given a weight of `since_last` and the last
/// TWAP, `from_start`. The new TWAP is then the weighted average of the two.
```
Where `since_last` and `from_start` would be equivalent to `weight_now` and `weight_last_timestamp` respectively

Reply:
Oh, you're **totally** right! 
I confused it while doing other changes. 
Already reverted it to the way it was before, thanks!